<!--
   Author: Russell Toris, Worcester Polytechnic Institute
  Version: March 3, 2015
-->

<launch>
  <!-- Grasp Database Connection Params -->
  <arg name="host" default="127.0.0.1" />
  <arg name="port" default="5432" />
  <arg name="user" default="ros" />
  <arg name="password" default="" />
  <arg name="db" default="graspdb" />

  <!-- General Grasp Collector Params -->
  <arg name="debug" default="false" />
  <arg name="robot_fixed_frame" default="base_footprint" />
  <arg name="grasp_frame" default="grasp_link" />

  <!-- Grasp Collector Action Server Client and Topic Params -->
  <arg name="gripper_action_server" default="/manipulation/gripper" />
  <arg name="lift_action_server" default="/manipulation/lift" />
  <arg name="verify_grasp_action_server" default="/manipulation/verify_grasp" />
  <arg name="segmented_objects_topic" default="/rail_segmentation/segmented_objects" />

  <!-- Main Node -->
  <node name="rail_grasp_collection" pkg="rail_grasp_collection" type="rail_grasp_collection" respawn="false" output="screen">
    <param name="host" value="$(arg host)" />
    <param name="port" value="$(arg port)" />
    <param name="user" value="$(arg user)" />
    <param name="password" value="$(arg password)" />
    <param name="db" value="$(arg db)" />
    <param name="debug" value="$(arg debug)" />
    <param name="robot_fixed_frame" value="$(arg robot_fixed_frame)" />
    <param name="grasp_frame" value="$(arg grasp_frame)" />
    <param name="gripper_action_server" value="$(arg gripper_action_server)" />
    <param name="lift_action_server" value="$(arg lift_action_server)" />
    <param name="verify_grasp_action_server" value="$(arg verify_grasp_action_server)" />
    <remap from="/rail_segmentation/segmented_objects" to="$(arg segmented_objects_topic)"/>
  </node>
</launch>
